src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509CertificateCollection.cs

```diff
@@ -173,68 +146,83 @@ public override int GetHashCode()
 
         public int IndexOf(X509Certificate value)
         {
-            return List.IndexOf(value);
+            return _list.IndexOf(value);
         }
 
         public void Insert(int index, X509Certificate value)
         {
-            List.Insert(index, value);
+            Requires.NotNull(value, "value");
+
+            _list.Insert(index, value);
         }
 
         public void Remove(X509Certificate value)
         {
-            List.Remove(value);
+            Requires.NotNull(value, "value");
+
+            _list.Remove(value);
         }
 
         public void RemoveAt(int index)
         {
-            if (index < 0 || index >= Count)
-                throw new ArgumentOutOfRangeException("index", SR.ArgumentOutOfRange_Index);
             _list.RemoveAt(index);
         }
 
         void ICollection.CopyTo(Array array, int index)
         {
-            ((ICollection)_list).CopyTo(array, index);
+            NonGenericList.CopyTo(array, index);
         }
 
-        int IList.Add(Object value)
+        int IList.Add(object value)
         {
-            if (value == null)
-                throw new ArgumentNullException("value");
+            Requires.NotNull(value, "value");
 
-            int index = _list.Count;
-            _list.Add(value);
-            return index;
+            return NonGenericList.Add(value);
         }
 
-        bool IList.Contains(Object value)
+        bool IList.Contains(object value)
         {
-            return _list.Contains(value);
+            return NonGenericList.Contains(value);
         }
 
-        int IList.IndexOf(Object value)
+        int IList.IndexOf(object value)
         {
-            return _list.IndexOf(value);
+            return NonGenericList.IndexOf(value);
         }
 
-        void IList.Insert(int index, Object value)
+        void IList.Insert(int index, object value)
         {
-            _list.Insert(index, value);
+            Requires.NotNull(value, "value");
+
+            NonGenericList.Insert(index, value);
         }
 
-        void IList.Remove(Object value)
+        void IList.Remove(object value)
         {
-            _list.Remove(value);
+            Requires.NotNull(value, "value");
+
+            NonGenericList.Remove(value);
+        }
+
+        private IList NonGenericList
+        {
+            get { return _list; }
         }
 
-        internal IList List
+        internal void GetEnumerator(out List<X509Certificate>.Enumerator enumerator)
         {
-            get { return this; }
+            enumerator = _list.GetEnumerator();
         }
 
-        private List<Object> _list;
-        private Object _syncRoot;
+        internal static class Requires
```

Comment:
I'm with @weshaggard on this one; I'm not sure the helper really adds much.  If there were a unified helper-for-throwing pattern that was consistent across corefx (or even the assembly) that'd be one thing; but this is something only consistent across these two classes.